### DISCUSSION TITLE
Chip size

### DISCUSSION THREAD

**Discussion by msinger** [emu-russia/dmgcpu#203] [General]
Hi,

I would like to add a scale and a measurement tool to my [map](http://iceboy.a-singer.de/dmg_cpu_b_map). Do you have a way to measure the width and height of the chip? The pictures from siliconpr0n have "20x" in their filenames, but I don't know how I could reliably calculate the size based on that. I would at least need the resolution for that.

**Comment by ogamespec** [comment]
Hi,

I'll try to think of something, but I don't have standard micrometer rulers, only by eye 

![image](https://user-images.githubusercontent.com/5828819/198736595-2924a17a-78a4-44af-8cc7-c99b24b5d725.png)

**Comment by ogamespec** [comment]
DMG Pads vs paper "dots"

![image](https://user-images.githubusercontent.com/5828819/198737958-5b78df92-1d6d-446d-b141-b51a17afef51.png)

**Comment by msinger** [comment]
Oh nice. So it's roughly 6 mm high. I never had any clue before. Can you use a caliper to get some numbers?

I noticed there are some markings at the top and left side of the chip. I was wondering if this some kind of ruler.

**Reply by ogamespec** [reply]
The caliper is very bad, with an error of +/- 100 µm. I don't know what it has to do with my Chinese caliper or the elasticity of the crystal.

I measured it now, it's about 6.8 x 6.1 mm. (+/- 100-200 µm, it's not very accurate)

**Reply by msinger** [reply]
Thank you. At least I have some numbers now. Didn't have anything before.

**Comment by ogamespec** [comment]
Generally I have seen special micrometer risks for such measurements, but I do not have anything like that in my home lab.

**Comment by ogamespec** [comment]
A micro ruler has arrived, I'll try to measure chip features with more resolution as I get my hands on it.

![image](https://github.com/user-attachments/assets/5d47fd4f-bf3e-499a-b9c8-19318ca8a976)

**Reply by msinger** [reply]
Oh, yes! That's great. I'm happy to change the scale on the map.

**Comment by ogamespec** [comment]
Measurement 1: dimensions of standard cells.
I used a 0.01mm scale, took 2 photos in focus: some cells and separately a photo of the ruler's grooves.
It is impossible to make a simultaneous focus of the chip features lying on the ruler - either the ruler or the chip will be defocused.
In photoshop I combined the 2 images, making the risks semi-transparent 50%.
It turned out like this:

![cell_dims](https://github.com/user-attachments/assets/928b01f0-1333-490d-b657-aa333e1583ec)

**Reply by msinger** [reply]
I just checked, I designed my cells to be ~52.3 µm wide, so it looks good, I think.

I was thinking about how to measure the whole chip: Would it be possible to put the calibration slide on top of the chip and make four images in total? Two on each corner of the chip with two different focuses, while not moving the chip while changing the focus. Could those images with different focus be overlaid in photoshop, so we can see both, the chip and the ruler? Of course, when moving the chip to look at the other edge, the ruler should be glued to it, so it doesn't move relatively to the chip.

**Reply by msinger** [reply]
I mean, sorry if this is a dumb question. I have no idea how it behaves when changing focus. Maybe it mechanically jiggles around too much.

**Reply by ogamespec** [reply]
I am not an optics expert, and school exercises are long forgotten, but it seems to me that 2 objects at different heights at different focal lengths will have different linear dimensions. The one closer to the lens (calibration slide) will have larger linear dimensions than the chip under it. Try moving your hand in front of your eye, I think you'll see what I mean :)

**Reply by ogamespec** [reply]
I'll take an overview photo of the x10 chip and overlay the slide lying next to it in photoshop.
The chip and the slide are about the same height, so the focal length will be the same. If not, I have some mica glasses, I'll put the chip next to it at the same height and check how much the linear dimensions of the ruler change compared to if the chip was just lying on the table / calibration ruler itself.

**Reply by ogamespec** [reply]
I'll read some more school physics textbook with a section on optics and come back :) It seems that there is no need to mess with a ruler in any particular way - it is enough to convert pixels to meters (or nanometers on our scale). Since the dimensions are linear, you can simply use the number of pixels to find out the dimensions of the required chip features :-)

**Reply by ogamespec** [reply]
https://www.youtube.com/watch?v=I8JR0LRMjPI

**Reply by msinger** [reply]
Yes, I understand, nearer objects are bigger. Though, the pixel at the center should stay in the same place, right?

I also understand, that we can use your measurement of the cell width and height to determine the size of other small objects on the chip. I'm just still interested in an exact width and height of the whole chip. For the browser map, I'm using your caliper measurements. I think that calculating the size of the chip based on the small cell measurements will result in a huge error, because you have to multiply it by a lot.

**Reply by ogamespec** [reply]
> Though, the pixel at the center should stay in the same place, right?

From the depths of ancestors' memory comes the word - parallax :)

> will result in a huge error

Aha I understand the need: you need to make sure that the addition of the small measurements will match the measurement of the whole chip (large measurement).
I will try to take pictures one of these days.
By the way, uneven stitching of the slides can also introduce error, so I will try to take a photo without stitching the slides together.
By the way, I have a 5x objective lying around somewhere, it will probably fit the whole chip, or at least most of it.

I also thought I'd try flipping the ruler with the stripe markings down so that they are close to the chip. I don't think the distance between the top surface of the chip and the paint would be too great and make a significant difference in focal length.

**Reply by msinger** [reply]
Good point about the stitching! Yes, even if the whole chip doesn't fit at once, the longer the measured distance, the smaller the measurement error should be, when we calculate the size of the chip from it.
